common/persistence/client/factory.go

```diff
@@ -322,18 +325,11 @@ func (f *factoryImpl) init(clusterName string, limiters map[string]quotas.Limite
 	f.datastores[storeTypeVisibility] = visibilityDataStore
 }
 
-func buildRatelimiters(cfg *config.Persistence) map[string]quotas.Limiter {
+func buildRatelimiters(cfg *config.Persistence, persistenceMaxQPS dynamicconfig.IntPropertyFn) map[string]quotas.Limiter {
 	result := make(map[string]quotas.Limiter, len(cfg.DataStores))
-	for dsName, ds := range cfg.DataStores {
-		qps := 0
-		if ds.Cassandra != nil {
-			qps = ds.Cassandra.MaxQPS
-		}
-		if ds.SQL != nil {
-			qps = ds.SQL.MaxQPS
-		}
-		if qps > 0 {
-			result[dsName] = quotas.NewSimpleRateLimiter(qps)
+	for dsName, _ := range cfg.DataStores {
+		if persistenceMaxQPS != nil {
+			result[dsName] = quotas.NewDynamicRateLimiter(func() float64 { return float64(persistenceMaxQPS()) })
```

Comment:
Check if the returned value from dynamic config is 0 before creating the rate limiter?

Reply:
What do you want to do if it is 0? I don't think we have these types of dynamic config checks for values in other places?

Reply:
Previous code did have a check on if the value is 0 right? I think if it's equal to 0, then no rate limiter should be created.

Reply:
Sorry, please ignore me. It was already checked in the if statement and I missed that part.